Sources/Currency/Currency.swift

```diff
@@ -0,0 +1,161 @@
+  /// For an amount that is likely to be used in "every day" usage, see ``roundedAmount``
+  var exactAmount: Decimal { get }
+
+  /// The cannonical representation of the ``exactAmount`` of money this value represents,
+  /// rounded using the `bankers` method to the currency's ``CurrencyDescriptor/minorUnits``.
+  ///
+  /// For example:
+  ///
+  ///     let usd = USD(10.007)
+  ///     let yen = JPY(100.9)
+  ///     let dinar = KWD(100.0019)
+  ///     print(usd, yen, dinar)
+  ///     // "USD(10.01), JPY(101), KWD(100.002)
+  var roundedAmount: Decimal { get }
+
+  /// Creates a representation of the currency with the exact value as given.
+  /// - Parameter exactAmount: The amount this instance should represent, without any rounding.
+  init(exactAmount: Decimal)
+
+  /// Creates a representation of the currency, converting the minor units representation to a decimal format.
+  ///
+  /// e.g. If you provide the value `100` for a currency with `2` minor units, the `exactAmount` will be `1.00`.
+  /// - Parameter minorUnits: The minor units this value will represent.
+  init(minorUnits: CurrencyMinorUnitRepresentation)
+
+  /// Rounds the ``exactAmount`` of money being represented using the given rounding method
+  /// to the currency's ``CurrencyDescriptor/minorUnits``.
+  /// - Parameter roundingMode: The desired rounding mode to use on the original ``exactAmount``.
+  /// - Complexity: O(1)
+  /// - Returns: The rounded amount.
+  func roundedAmount(using roundingMode: NSDecimalNumber.RoundingMode) -> Decimal
+}
+
+// MARK: Defaults
+
+extension Currency {
+  public var minorUnits: CurrencyMinorUnitRepresentation {
+    let scaledAmount = self.exactAmount * Self.descriptor.minorUnitsCoefficient(for: .exactAmount)
+    return .init(scaledAmount.int64Value)
+  }
+
+  public var roundedAmount: Decimal { self.roundedAmount(using: .bankers) }
+
+  public init(minorUnits: CurrencyMinorUnitRepresentation) {
+    let amount = Decimal(minorUnits) * Self.descriptor.minorUnitsCoefficient(for: .minorUnits)
+    self.init(exactAmount: amount)
+  }
+
+  public func roundedAmount(using roundingMode: NSDecimalNumber.RoundingMode) -> Decimal {
+    var sourceAmount = self.exactAmount
+    var result = Decimal.zero
+    NSDecimalRound(&result, &sourceAmount, .init(Self.descriptor.minorUnits), roundingMode)
+    return result
+  }
+}
+
+extension Currency where Self: CurrencyDescriptor {
+  public static var descriptor: CurrencyDescriptor.Type { Self.self }
+}
+
+// MARK: Equatable
+
+extension Currency {
+  public static func ==<Other: Currency>(lhs: Self, rhs: Other) -> Bool {
+    guard Self.descriptor == Other.descriptor else { return false }
+    return lhs.exactAmount == rhs.exactAmount
+  }
+}
+
+// MARK: Comparable
+
+extension Currency {
+  public static func < <Other: Currency>(lhs: Self, rhs: Other) -> Bool {
+    guard Self.descriptor == Other.descriptor else {
+      return Self.descriptor.alphabeticCode < Other.descriptor.alphabeticCode
```

Comment:
❔ Should we callout somewhere that comparing two different currencies uses the alphabetic code comparison. Some people could think it will do the exchange rate from one currency to another one for the operation but that is not the case.

Reply:
Did you mean to comment on another line? For comparisons this is totally fine

After we nail down all of the internals, I will be doing a pass on the documentation.

Reply:
There’s no change to the value, comparison is just a sorting operation

Reply:
I meant that function. It is not intuitive to me that comparing two different currencies will use the currency alphabetic code and therefore I thought it will be good to have it in the documentation.

Reply:
I'll make sure to call it out specifically that the currency will sort according to the alphabetic identifier.